src/extension.ts

```diff
@@ -33,24 +41,49 @@ class PSCompletionItem extends CompletionItem {
 }
 
 class PSCompletionProvider implements CompletionItemProvider {
-  constructor(private snippets: [Snippet]) {}
+  constructor(private snippets: Snippets) {}
 
   provideCompletionItems(): ProviderResult<CompletionItem[]> {
     return Object.keys(snippets).map(key => new PSCompletionItem(snippets[key]))
   }
 }
 
 export async function activate(context: ExtensionContext) {
-  let registerCompletionProvider = languages.registerCompletionItemProvider(
-    'javascript',
-    new PSCompletionProvider(snippets as [Snippet]),
-    '*'
-  )
+  async function workspaceFoldersChange(folders: WorkspaceFolder[]) {
+    let config
+    if (folders) {
+      config = await prettier.resolveConfig(folders[0].uri.path)
+    } else {
+      config = workspace.getConfiguration(PRETTIER_EXTENSION)
```

Comment:
This line makes the extension dependent on `prettier-vscode` too. Is it what we want?

Reply:
That was intentional. Do you think it's wrong?

Reply:
It still works even if you don't have it installed. Just throws some warnings into the console since we're not passing `PrettierConfig`. As far as depending on Prettier form VS Marketplace, it should not be enforced. The extension can work out of the box. Maybe as recommended dependency.